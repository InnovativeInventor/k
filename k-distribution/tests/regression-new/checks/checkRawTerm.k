// Copyright (c) K Team. All Rights Reserved.

module CHECKRAWTERM-SYNTAX
  syntax Foo ::= foo() | bar()
endmodule

module CHECKRAWTERM
  imports CHECKRAWTERM-SYNTAX

  rule foo() => #rawTerm(foo())
  rule #rawTerm(foo()) => bar()
endmodule
