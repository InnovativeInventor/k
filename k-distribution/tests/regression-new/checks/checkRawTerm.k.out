[Error] Compiler: Found usage of #rawTerm; this production is reserved for K backend usage and cannot appear in user code.
	Source(checkRawTerm.k)
	Location(10,17,10,32)
	10 |	  rule foo() => #rawTerm(foo())
	   .	                ^~~~~~~~~~~~~~~
[Error] Compiler: Found usage of #rawTerm; this production is reserved for K backend usage and cannot appear in user code.
	Source(checkRawTerm.k)
	Location(11,8,11,23)
	11 |	  rule #rawTerm(foo()) => bar()
	   .	       ^~~~~~~~~~~~~~~
[Error] Compiler: Had 2 structural errors.
